pkg/ns/ns.go

```diff
@@ -95,33 +107,29 @@ func IsNS(nspath string) (isNS bool, msg string, err error) {
 		validPathContent := "ns/"
 		validName := strings.Contains(nspath, validPathContent)
 		if !validName {
-			msg = fmt.Sprintf("path doesn't contain %q", validPathContent)
-			return
+			return NSPathNotNSErr{msg: fmt.Sprintf("path %q doesn't contain %q", nspath, validPathContent)}
 		}
-		isNS = true
+
+		return nil
 	case NSFS_MAGIC:
 		// Kernel >= 3.19
 
-		isNS = true
+		return nil
 	default:
-		msg = fmt.Sprintf("unknown FS magic: %x", stat.Type)
+		return NSPathNotNSErr{msg: fmt.Sprintf("unknown FS magic on %q: %x", nspath, stat.Type)}
 	}
-	return
 }
 
 // Returns an object representing the namespace referred to by @path
 func GetNS(nspath string) (NetNS, error) {
-	isNS, msg, err := IsNS(nspath)
+	err := IsNSorErr(nspath)
 	if err != nil {
 		return nil, err
 	}
-	if !isNS {
-		return nil, fmt.Errorf("no network namespace detected on %s: %s", nspath, msg)
-	}
 
 	fd, err := os.Open(nspath)
 	if err != nil {
-		return nil, fmt.Errorf("Failed to open %v: %v", nspath, err)
+		return nil, err
```

Comment:
On second thought, this is not really something I'd like to consider. There is also a small window between the call to `IsNSorErr()` and the actual return of that error value, but yet it doesn't make sense to check it again. We're talking about very short periods of time, when the error we're probably trying to catch is in time length of a reboot or similar.